rlgym_tools/sb3_utils/sb3_log_reward.py

```diff
@@ -1,44 +1,152 @@
+        self.dumpfile = f'bin/combinedlogfiles/{self.index}.txt'
+
+        # clean out any files that are left over from previous runs
+        if self.index == 0:
+            for file in os.listdir('bin/combinedlogfiles'):
+                if '.txt' in file:
+                    os.remove('bin/combinedlogfiles/'+file)
+
+
+        # initiates the array that will store the rewards
+        self.returns = np.zeros(len(self.reward_functions))
+
+    def reset(self, initial_state: GameState):
+        self.returns = np.zeros(len(self.reward_functions))
+        super().reset(initial_state)
+
+    def get_reward(self, player: PlayerData, state: GameState, previous_action: np.ndarray) -> float:
+        rewards = [
+            func.get_reward(player, state, previous_action)
+            for func in self.reward_functions
+        ]
+
+        self.returns += [a*b for a, b in zip(rewards, self.reward_weights)] # store the rewards
+
+        return float(np.dot(self.reward_weights, rewards))
+
+    def get_final_reward(self, player: PlayerData, state: GameState, previous_action: np.ndarray) -> float:
+        rewards = [
+            func.get_final_reward(player, state, previous_action)
+            for func in self.reward_functions
+        ]
+
+        self.returns += [a*b for a, b in zip(rewards, self.reward_weights)] # store the rewards
+
+        # write the rewards to file and reset
+        with open(self.dumpfile, 'a') as f:
+            # in a list so we can eval to get it back easily
+            f.write('\n' + str(list(self.returns)))
+        self.returns = np.zeros(len(self.reward_functions))
+
+        return float(np.dot(self.reward_weights, rewards))
+
+
+class SB3CombinedLogRewardCallback(BaseCallback):
+    def __init__(self, rew_names:List[str]):
+        """
+        Callback to log the data from a SB3CombinedLogReward to the
+        same log as the model.
+
+        :param rew_names: List of names that the logger will use for
+        each reward.
+        """
+        super().__init__()
+        self.rew_names = rew_names
+
+    def _on_step(self) -> bool:
+        return True
+
+    def _on_rollout_end(self) -> None:
+        returns = []
+        # get all returns out of the files
+        for file in os.listdir('bin/combinedlogfiles'):
+            if '.txt' in file:
+                with open('bin/combinedlogfiles/'+file, 'r') as f:
+                    for line in f:
+                        if line != '\n':
+                            returns.append(eval(line.strip()))
+                # empty the file ready for the next rollout
+                with open('bin/combinedlogfiles/'+file, 'w') as f:
+                    f.write('') 
+
+        returns = np.array(returns)
+        # use as many names as provided, then call them by their index after that
+        names = [self.rew_names[i] if i < len(self.rew_names) else f'reward_{i}' for i in range(returns.shape[1])]

```

Comment:
I don't think the creation of this list helps.

Would probably be clearer if put within the loop below:
```python
        for i, name in enumerate(names):
            try:
                name = self.reward_names[i]
            except IndexError:
                name = f"reward_{i}"
            self.model.logger.record('rewards/' + name, np.mean(returns[:, i]))
```